tests/conftest.py

```diff
@@ -38,3 +42,34 @@ def caplog(caplog: LogCaptureFixture):
     )
     yield caplog
     logger.remove(handler_id)
+
+
+@pytest.fixture
+def test_specific_results_dir():
+    """Creates a temporary and unique directory for each test to store results.
+
+    This fixture yields the path for the test to use and then cleans up the directory
+    after the test is done.
+
+    Notes
+    -----
+    We cannot use pytest's tmp_path fixture because other nodes do not have access to it.
+    Also, do not use a context manager (i.e. tempfile.TemporaryDirectory) because it's too
+    difficult to debug when the test fails b/c the dir gets deleted.
+    """
+    results_dir = tempfile.mkdtemp(dir=RESULTS_DIR)
+    # give the dir the same permissions as the parent directory so that cluster jobs
+    # can write to it
+    os.chmod(results_dir, os.stat(RESULTS_DIR).st_mode)
+    results_dir = Path(results_dir)
+    yield results_dir
+
+    # Try 10 to delete the dir.
+    # NOTE: There seems to be times where the directory is not removed (even after
```

Comment:
I can't for the life of me figure out what process is spawning these extra tmpdirs. I think it's spark?

Reply:
I think this might be a result of the network file system: if Spark hasn't yet given up its _file handle_ on the dir, it won't let you delete. https://unix.stackexchange.com/questions/127507/delete-files-folders-with-nfs-files